File: Algorithms/01-Warmup/06-staircase/06.cpp
#include <iostream>

void staircase(int n) {
    for (int i = 1; i <= n; i++) {
        std::string hash = "";
        std::string space = "";
        for (int j = 1; j <= i; j++)  {
            hash += "#";
        }
        for (int k = n; k > i; k--) {
            space += " ";
        }
        std::cout << space << hash << std::endl;
    }
}

int main() {
    int n;
    std::cin >> n;
    staircase(n);
    return 0;
}

File: Algorithms/01-Warmup/06-staircase/06.java
import java.util.Scanner;

class Solution {

    static void staircase(int n) {
        for (int i = 1; i <= n; i++) {
            String space = "";
            String hash = "";
            for (int j = 1; j <= i; j++) {
                hash += "#";
            }
            for (int k = n; k > i; k--) {
                space += " ";

            }
            System.out.printf("%s%s\n", space, hash);
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        staircase(n);
        in.close();
    }
}
